src/api/clubsMembers.ts

```diff
@@ -0,0 +1,110 @@
+import { MemberType, MembershipType } from '../model/Member';
+import { getSupabaseClient } from './supabase';
+
+export const isUserClubMember = async (userId: string, clubId: number) => {
+  try {
+    const { data, error } = await getSupabaseClient()
+      .from('clubs_members')
+      .select<string, MembershipType>()
+      .eq('memberId', userId)
+      .eq('clubId', clubId);
+
+    if (error) {
+      throw error;
+    }
+
+    return data.length > 0;
+  } catch (error) {
+    console.error('Error checking club membership:', error);
+    throw error;
+  }
+};
+
+export const inviteMemberByEmail = async (clubId: number, email: string) => {
+  try {
+    const { data: userData } = await getSupabaseClient()
+      .from('members')
+      .select<string, MemberType>('id')
+      .eq('email', email)
+      .single();
+
+    if (!userData) {
+      throw new Error('User with this email not found');
+    }
+
+    const isMember = await isUserClubMember(userData.id, clubId);
+    if (isMember) {
+      throw new Error('User is already a member of this club');
+    }
```

Comment:
When will we be scaling the application for the thousands of nerds using it? This might be a good place to start with "make it performant". Make it only 1 DB call

For now, keep it.